src/connector.js

```diff
@@ -0,0 +1,769 @@
+	}
+	
+	searchBoxController.updateText( qString );
+	searchBoxController.submit();
+}
+
+if ( hashParams.q && searchBoxElement ) {
+	searchBoxElement.value = hashParams.q;
+}
+else if ( urlParams.q && searchBoxElement ) {
+	searchBoxElement.value = urlParams.q;
+}
+
+// Get the query portion of the URL
+const fragment = () => {
+	const hash = window.location.hash.slice( 1 );
+	if (!statusController.state.firstSearchExecuted && !hashParams.q ) {					
+		return window.location.search.slice( 1 ); // use query string if hash is empty
+	}
+	
+	return hash;
+};
+
+const urlManager = buildUrlManager( headlessEngine, {
+	initialState: {
+		fragment: fragment(),
+	},
+} );
+
+// Sync controllers when URL changes
+let updateSearchBoxFromState = false;
+const onHashChange = () => { 
+	updateSearchBoxFromState = true;
+	urlManager.synchronize( fragment() );
+};
+
+// Execute a search if parameters in the URL on page load
+if ( !statusController.state.firstSearchExecuted && fragment() ) {
+	headlessEngine.executeFirstSearch();
+}
+
+// Listen to URL change (hash)
+window.addEventListener( 'hashchange', onHashChange );
+
+// Unsubscribe to controllers ****** Waiting on Coveo to get updated version
+urlManager.subscribe( () => {
+	if ( !params.enableHistoryPush || window.location.origin.startsWith( 'file://' ) ) {
+		return;
+	}
+
+	let hash = `#${urlManager.state.fragment}`;
+
+	if ( !statusController.state.firstSearchExecuted ) {
+		window.history.replaceState( null, document.title, window.location.origin + winPath + hash );
+	} else {
+		window.history.pushState( null, document.title, window.location.origin + winPath + hash );
+	}
+} );
+
+let searchBoxState;
+let resultListState;
+let querySummaryState;
+let didYouMeanState;
+let pagerState;
+let lastCharKeyUp;
+
+// Show query suggestions if a search action was not executed (if enabled)
+function updateSearchBoxState( newState ) {
+	const previousState = searchBoxState;
+	searchBoxState = newState;
+	
+	if ( updateSearchBoxFromState && searchBoxElement && searchBoxElement.value !== newState.value ) {
+		searchBoxElement.value = newState.value;
+		updateSearchBoxFromState = false;
+		return;
+	}
+	
+	if ( !suggestionsElement ) {
+		return;
+	}
+		
+	if ( lastCharKeyUp === 13 ) {
+		suggestionsElement.style.display = 'none';
+		return;
+	}
+
+	if ( !searchBoxState.isLoadingSuggestions && previousState?.isLoadingSuggestions ) {
+		suggestionsElement.textContent = '';
+		searchBoxState.suggestions.forEach( ( suggestion ) => {
+			const node = document.createElement( "li" );
+			node.setAttribute( "class", "suggestion-item" );
+			node.onclick = ( e ) => { 
+				searchBoxController.selectSuggestion(e.currentTarget.innerText);
+				searchBoxElement.value = e.currentTarget.innerText;
+			};
+			node.innerHTML = suggestion.highlightedValue;
+			suggestionsElement.appendChild( node );
+		});
+		
+		if ( searchBoxState.suggestions.length > 0 ) {
+			suggestionsElement.style.display = 'block';
```

Comment:
Why to not use the `hidden` property of an DOM element instead of inline style?

```suggestion
			suggestionsElement.hidden = false;
```